packages/server/lib/controllers/runner.ts

```diff
@@ -1,101 +1,11 @@
-import _ from 'lodash'
 import type { Request, Response } from 'express'
 import send from 'send'
-import os from 'os'
-import { fs } from '../util/fs'
-import path from 'path'
-import Debug from 'debug'
-import pkg from '@packages/root'
-import { getPathToDist, getPathToIndex, RunnerPkg } from '@packages/resolve-dist'
-import type { InitializeRoutes } from '../routes'
-import type { PlatformName } from '@packages/types'
-import type { Cfg } from '../project-base'
-
-const debug = Debug('cypress:server:runner')
-
-const PATH_TO_NON_PROXIED_ERROR = path.join(__dirname, '..', 'html', 'non_proxied_error.html')
-
-const _serveNonProxiedError = (res: Response) => {
-  return fs.readFile(PATH_TO_NON_PROXIED_ERROR)
-  .then((html) => {
-    return res.type('html').end(html)
-  })
-}
-
-export interface ServeOptions extends Pick<InitializeRoutes, 'getSpec' | 'config' | 'getCurrentBrowser' | 'remoteStates' | 'exit'> {
-  testingType: Cypress.TestingType
-}
-
-export const serveRunner = (runnerPkg: RunnerPkg, config: Cfg, res: Response) => {
-  // base64 before embedding so user-supplied contents can't break out of <script>
-  // https://github.com/cypress-io/cypress/issues/4952
-  const base64Config = Buffer.from(JSON.stringify(config)).toString('base64')
-
-  const runnerPath = process.env.CYPRESS_INTERNAL_RUNNER_PATH || getPathToIndex(runnerPkg)
-
-  // Chrome plans to make document.domain immutable in Chrome 106, with the default value
-  // of the Origin-Agent-Cluster header becoming 'true'. We explicitly disable this header
-  // so that we can continue to support tests that visit multiple subdomains in a single spec.
-  // https://github.com/cypress-io/cypress/issues/20147
-  res.setHeader('Origin-Agent-Cluster', '?0')
-
-  return res.render(runnerPath, {
-    base64Config,
-    projectName: config.projectName,
-    namespace: config.namespace,
-  })
-}
+import { getPathToDist } from '@packages/resolve-dist'
 
 export const runner = {
-  serve (req, res, runnerPkg: RunnerPkg, options: ServeOptions) {
```

Comment:
Did not realize this was unused... I wonder if this *should* live in data-context? Server seems like a better place for, well, serving things. The line between server/data-context is really blurred, though. I think the goal was incrementally moving things so we could strongly type them, but a lot of things ended up just getting dumped there (probably since it has more strict types, so it's easier to work in).

Reply:
yeah the line between the data-context & server is very blurred. I think intention was to separate server vs config/setup type concepts but we didn't get all the way there how intertwined some of this is